libproxy/url.cpp

```diff
@@ -499,20 +498,33 @@ char* url::get_pac() {
 			if (content_length >= PAC_MAX_SIZE) break;
 
 			while (content_length == 0 || recvd != content_length) {
-				int r = recv(sock, buffer + recvd,
-				             content_length == 0 ? PAC_HTTP_BLOCK_SIZE
-				                                 : content_length - recvd, 0);
+				// Calculate length to recv
+				unsigned int length_to_read = content_length == 0 ? PAC_HTTP_BLOCK_SIZE
```

Comment:
As you're touching this already, can you add parentheses around `content_length - recvd`?